ckanext/validation/validation_status_helper.py

```diff
@@ -0,0 +1,159 @@
+
+        If in state 'created' or 'running' and created time stamp within last hour:
+            raise exception ValidationJobAlreadyEnqueued
+
+        Else: (object exists and is in final state(success, failure, error))
+            reset record to clean state with status 'created', created with timestamp now
+
+
+        :param self:
+        :param string resource_id: resource_id of job
+        :return Validation record
+        :throws ValidationJobAlreadyEnqueued exception
+
+        '''
+        log.debug("createValidationJob: %s", resource_id)
+        validationRecord = self.getValidationJob(session, resource_id)
+
+        if validationRecord is not None:
+            status = validationRecord.status
+
+            if (status == StatusTypes.running or status == StatusTypes.created) and self.getHoursSince(validationRecord.created) < 1:
+                error_message = "Validation Job already in pending state: {} on resource: {} created on (gmt): {}"\
+                    .format(validationRecord.status, resource_id, validationRecord.create.isoDateTime())
+                log.error(error_message)
+                raise ValidationJobAlreadyEnqueued(error_message)
+
+        if validationRecord is None:
+            validationRecord = model.Validation(resource_id=resource_id)
+
+        validationRecord.finished = None
+        validationRecord.report = None
+        validationRecord.error = None
+        validationRecord.created = datetime.datetime.utcnow()
+        validationRecord.status = StatusTypes.created
+
+        session.add(validationRecord)
+        session.commit()
+        session.flush()
+        return validationRecord
+
+    def updateValidationJobStatus(self, session=None, resource_id=None, status=None, report=None, error=None):
+        # type: (object, Session, str, str, object, object) -> model.Validation
+        """
+        If report or error is attached, update finished to be now
+        :param self:
+        :param session Session
+        :param resource_id:
+        :param status:
+        :param report:
+        :param error:
+        :return:
+        """
+        log.debug("updateValidationJobStatus: %s status: %s", resource_id, status)
+        validationRecord = self.getValidationJob(session, resource_id)
+
+        if validationRecord is None:
+            log.error("record not found to update statues: %s", resource_id)
+            raise ValidationJobDoesNotExist()
+
+        # Handle already running status in in last hour
+        if status == StatusTypes.running and validationRecord.status == status:
+            if self.getHoursSince(validationRecord.created) < 1:
+                raise ValidationJobAlreadyRunning()
+
+        validationRecord.status = status
+        validationRecord.report = report
+        validationRecord.error = error
+        if status == StatusTypes.success or status == StatusTypes.failure or status == StatusTypes.error:
```

Comment:
Could write this as `if status in [StatusTypes.success, StatusTypes.failure, StatusTypes.error]:`